lib/matplotlib/axis.py

```diff
@@ -1225,15 +1227,19 @@ def _set_lim(self, v0, v1, *, emit=True, auto):
         if auto is not None:
             self._set_autoscale_on(bool(auto))
 
+        # Undocumented internal feature: setting emit to _NoRecursionMarker causes
+        # signal emission but doesn't propagate to shared axed (thus preventing
+        # infinite recursion).
         if emit:
             self.axes.callbacks.process(f"{name}lim_changed", self.axes)
-            # Call all of the other axes that are shared with this one
-            for other in self._get_shared_axes():
-                if other is not self.axes:
-                    other._axis_map[name]._set_lim(
-                        v0, v1, emit=False, auto=auto)
```

Comment:
could this just be solved with the one line change of:

```python
# Emit callbacks on shared axes without recursing
other.callbacks.process(f"{name}lim_changed", other)
```

inside of this for loop/if condition

seems overdesigned to introduce undocumented sentinel behavior to get one line to run.

Reply:
Ah yes, that's *much* better.